webrtc.html

```diff
@@ -11209,6 +11205,42 @@ <h2>
                     </p>
                   </li>
                 </ol>
+                <p>
+                  The <dfn class="export">codec match</dfn> algorithm given two {{RTCRtpCodec}} or compatible dictionaries <var>first</var> and <var>second</var> is as follows:
+                </p>
+                <ol class=algorithm>
+                  <li>
+                    <p>
+                      If <var>first</var>.{{RTCRtpCodec/mimeType}} is different from
+                      <var>second</var>.{{RTCRtpCodec/mimeType}}, return <code>false</code>.
+                    </p>
+                  </li>
+                  <li>
+                    <p>
+                      If <var>first</var>.{{RTCRtpCodec/clockRate}} is different from
+                      <var>second</var>.{{RTCRtpCodec/clockRate}}, return <code>false</code>.
+                    </p>
+                  </li>
+                  <li>
+                    <p>
+                      If <var>first</var>.{{RTCRtpCodec/channels}} and <var>second</var>.{{RTCRtpCodec/channels}}
+                      are not both missing or if <var>first</var>.{{RTCRtpCodec/channels}} is different from
+                      <var>second</var>.{{RTCRtpCodec/channels}}, return <code>false</code>.
```

Comment:
A codec match algorithm is also different for video codecs vs audio codecs (channels should be ignored on input when matching video codecs.

So we might need to pass in _kind_. Once we do that, it's tempting to bake in the list of codecs and call it a valid codec algorithm, given _kind_.

I've also opened https://github.com/w3c/webrtc-pc/issues/2849 to add internal algorithms we can call.

Reply:
> A codec match algorithm is also different for video codecs vs audio codecs (channels should be ignored on input when matching video codecs).

The algorithm is not different. But video codecs never have a `channels` member, so they can never match any provided value that have one. If you provide a `channels` value, it will return false.